utensor_cgen/transformer/tflite_exporter.py

```diff
@@ -0,0 +1,412 @@
+      ugraph.output_nodes[i] = name_new
+      break
+
+def get_fullyconnected_builtin_option(fbuilder, op_info):
+
+  weight_format = tflite.FullyConnectedOptionsWeightsFormat.FullyConnectedOptionsWeightsFormat.DEFAULT
+
+  tflite.FullyConnectedOptions.FullyConnectedOptionsStart(fbuilder)
+  #FIXME: node fusion and select an activation function here
+  tflite.FullyConnectedOptions.FullyConnectedOptionsAddFusedActivationFunction(fbuilder, ActivationFunctionType.NONE)
+  tflite.FullyConnectedOptions.FullyConnectedOptionsAddWeightsFormat(fbuilder, weight_format)
+  #FIXME: check the parameter here
+  tflite.FullyConnectedOptions.FullyConnectedOptionsAddKeepNumDims(fbuilder, False)
+  obj = tflite.FullyConnectedOptions.FullyConnectedOptionsEnd(fbuilder)
+
+  return obj, BuiltinOptions.FullyConnectedOptions
+
+def get_add_builtin_option(fbuilder, op_info):
+
+  tflite.AddOptions.AddOptionsStart(fbuilder)
+  #FIXME: node fusion and select an activation function here
+  tflite.AddOptions.AddOptionsAddFusedActivationFunction(fbuilder, ActivationFunctionType.NONE)
+  obj = tflite.AddOptions.AddOptionsEnd(fbuilder)
+
+  return obj, BuiltinOptions.AddOptions
+
+def tensor_type_lookup(numpy_dtype):
+  TensorType = tflite.TensorType.TensorType
+  lookup_map = dict()
+  lookup_map[np.dtype('float32')] = TensorType.FLOAT32
+  lookup_map[np.dtype('int8')] = TensorType.INT8
+
+  return lookup_map[numpy_dtype]
+
+
+class FlatbufferOpManager:
+    op_list = list()
+    code_name_lookup = {v: k for k, v in BuiltinOperator.__dict__.items()}
+    
+    def regsiter_op(self, op_type_str):
+        if op_type_str not in BuiltinOperator.__dict__:
+          assert("invalid op str")
+        assert "op already been registered", not self.is_registered(op_type_str)
+        self.op_list.append(op_type_str)
+        return self.op_code(op_type_str)
+
+    def is_registered(self, op_type_str):
+      return op_type_str in self.op_list
+    
+    def op_index(self, op_type_str):
+        return self.op_list.index(op_type_str)
+    def op_code(self, op_type_str):
+        return BuiltinOperator.__dict__[op_type_str]
+    def code2name(self, _op_code):
+        return self.code_name_lookup[_op_code]
+    def code2index(self, _op_code):
+        return self.op_list.index(self.code2name(_op_code))
+    def index2code(self, _index):
+        op_type = self.op_list[_index]
+        return  BuiltinOperator.__dict__[op_type]
+    def index2name(self, _index):
+        code = self.index2code(_index)
+        return self.code2name(code)
+
+class TFLiteExporter(Transformer):
+  METHOD_NAME = 'tflite_export'
+  KWARGS_NAMESCOPE = '_tflite_export'
+  __Max_fbuff_size = 1024 * 10
+  static_op_types = ["Inline", "Const"]
+  schema_version = 3
+  file_ident = "TFL3"
+
+  def __init__(self, input_tensors, output_tensors):
+    """
+    input_tensors: a list of input tensor names
+    output_tensors: alist of output tensor names
+    """
+
+    self.prune_graph = False #what is this?
```

Comment:
I did some trick on the `Transformer`.
And setting this flag to `False` simply to skip a pruning phase.
Or all transformed will go through a pruning phase to check if there is any op that is no longer used in the graph and it will remove such ops.